# Python Pandas Projektek | Nemzetközi szállítási adatok elemzése
---

![](logistics.jpg)

*Kép forrása: pixabay*

## 1. Az adatok betöltése és az alapadatok megjelenítése

### 1.1. Az adatok betöltése Pandas DataFrame-be
A `szallitas.csv` nevű fájl adatait betöltjük a Pandas DataFrame-be. 

Létrehozunk egy változót `szallitas` néven és megjelenítjük az adatkeretet. 

In [1]:
import pandas as pd

szallitas = pd.read_csv("szallitas.csv")

szallitas

azonosito  felado_orszag címzett_orszag termek_kategoria  suly_kg  \
0        101           Kína   Magyarország      Elektronika      5.2   
1        102    Németország   Magyarország          Ruházat      1.5   
2        103            USA  Franciaország      Elektronika     12.8   
3        104           Kína            USA          Ruházat      3.1   
4        105    Németország    Olaszország       Élelmiszer      8.7   
5        106           Kína   Magyarország      Elektronika     25.4   
6        107            USA   Magyarország          Ruházat      4.0   
7        108  Franciaország            USA       Élelmiszer      7.5   

   szallitasi_ido_nap  
0                  14  
1                   4  
2                  21  
3                  18  
4                   6  
5                  16  
6                  10  
7                  12

### 1.2. Az adatkeret első 5 sorának megjelenítése

In [2]:
szallitas.head()

azonosito felado_orszag címzett_orszag termek_kategoria  suly_kg  \
0        101          Kína   Magyarország      Elektronika      5.2   
1        102   Németország   Magyarország          Ruházat      1.5   
2        103           USA  Franciaország      Elektronika     12.8   
3        104          Kína            USA          Ruházat      3.1   
4        105   Németország    Olaszország       Élelmiszer      8.7   

   szallitasi_ido_nap  
0                  14  
1                   4  
2                  21  
3                  18  
4                   6

### 1.3. A hiányzó adatok (NaN) ellenőrzése

In [3]:
szallitas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   azonosito           8 non-null      int64  
 1   felado_orszag       8 non-null      object 
 2   címzett_orszag      8 non-null      object 
 3   termek_kategoria    8 non-null      object 
 4   suly_kg             8 non-null      float64
 5   szallitasi_ido_nap  8 non-null      int64  
dtypes: float64(1), int64(2), object(3)
memory usage: 516.0+ bytes


## 2. Az adatok szűrése

### 2.1. Az összes szállítmány kiválasztása, amely Magyarországra érkezett

In [4]:
szallitas[szallitas["címzett_orszag"] == "Magyarország"]

azonosito felado_orszag címzett_orszag termek_kategoria  suly_kg  \
0        101          Kína   Magyarország      Elektronika      5.2   
1        102   Németország   Magyarország          Ruházat      1.5   
5        106          Kína   Magyarország      Elektronika     25.4   
6        107           USA   Magyarország          Ruházat      4.0   

   szallitasi_ido_nap  
0                  14  
1                   4  
5                  16  
6                  10

### 2.2. A több, mint 10 kg súlyú szállítmányok kiszűrése

In [5]:
szallitas[szallitas["suly_kg"] > 10]

azonosito felado_orszag címzett_orszag termek_kategoria  suly_kg  \
2        103           USA  Franciaország      Elektronika     12.8   
5        106          Kína   Magyarország      Elektronika     25.4   

   szallitasi_ido_nap  
2                  21  
5                  16

### 2.3. A Kínából induló, `Elektronika` kategóriájú szállítmányok megjelenítése

In [6]:
szallitas[(szallitas["felado_orszag"] == "Kína") & (szallitas["termek_kategoria"] == "Elektronika")]

azonosito felado_orszag címzett_orszag termek_kategoria  suly_kg  \
0        101          Kína   Magyarország      Elektronika      5.2   
5        106          Kína   Magyarország      Elektronika     25.4   

   szallitasi_ido_nap  
0                  14  
5                  16

## 3. Csoportosítás és aggregáció

### 3.1. A szállítmány megjelenítése ország és mennyiség szerint
Ebben az esetben ki kell számítani, hogy melyik országból mennyi szállítmány érkezett. 

Ehhez elöször csoportosítani kell az országokat a `gropby()` metódussal a `címzett_orszag` oszlop szerint.

Ezután az `azonosito` alapján a `count()` metódus segítségével meg kell számolni az elemeket.

In [7]:
szallitas.groupby("címzett_orszag")["azonosito"].count()

címzett_orszag
Franciaország    1
Magyarország     4
Olaszország      1
USA              2
Name: azonosito, dtype: int64

**A kimenet:**

    címzett_orszag
    Franciaország    1
    Magyarország     4
    Olaszország      1
    USA              2
    Name: azonosito, dtype: int64

### 3.2. Az átlagos szállítási idő meghatározása a termék kategória szerint

In [8]:
szallitas.groupby("termek_kategoria")["szallitasi_ido_nap"].mean()

termek_kategoria
Elektronika    17.000000
Ruházat        10.666667
Élelmiszer      9.000000
Name: szallitasi_ido_nap, dtype: float64

**A kimenet:**

    termek_kategoria
    Elektronika    17.000000
    Ruházat        10.666667
    Élelmiszer      9.000000
    Name: szallitasi_ido_nap, dtype: float64

### 3.3. Az egyes feladó országokból indított szállítmányok teljes súlya

In [9]:
szallitas.groupby("felado_orszag")["suly_kg"].sum()

felado_orszag
Franciaország     7.5
Kína             33.7
Németország      10.2
USA              16.8
Name: suly_kg, dtype: float64

**A kimenet:**

    felado_orszag
    Franciaország     7.5
    Kína             33.7
    Németország      10.2
    USA              16.8
    Name: suly_kg, dtype: float64

## 4. Új oszlop hozzáadása és adatmanipuláció

### 4.1. Új oszlop létrehozása `suly_lb` néven, amely a súlyt kilogrammból fontba konvertálja
Létrehozzuk a DataFrame egy új oszlopát úgy, hogy megadjuk a nevét szögletes zárójelben, majd hozzárendeljük az értékeit.

> 1 kg = 2.20462 lb

In [10]:
szallitas["suly_lb"] = szallitas["suly_kg"] * 2.20462

Nézzük meg az új oszlopot a `head()` metódussal.

In [11]:
szallitas.head()

azonosito felado_orszag címzett_orszag termek_kategoria  suly_kg  \
0        101          Kína   Magyarország      Elektronika      5.2   
1        102   Németország   Magyarország          Ruházat      1.5   
2        103           USA  Franciaország      Elektronika     12.8   
3        104          Kína            USA          Ruházat      3.1   
4        105   Németország    Olaszország       Élelmiszer      8.7   

   szallitasi_ido_nap    suly_lb  
0                  14  11.464024  
1                   4   3.306930  
2                  21  28.219136  
3                  18   6.834322  
4                   6  19.180194

### 4.2. Új oszlop létrehozása `gyors_szallitas` néven, ami True értéket ad, ha a `szallitasi_ido_nap` kisebb, mint 7 nap

In [12]:
szallitas["gyors_szallitas"] = szallitas["szallitasi_ido_nap"] < 7 

In [13]:
szallitas.head()

azonosito felado_orszag címzett_orszag termek_kategoria  suly_kg  \
0        101          Kína   Magyarország      Elektronika      5.2   
1        102   Németország   Magyarország          Ruházat      1.5   
2        103           USA  Franciaország      Elektronika     12.8   
3        104          Kína            USA          Ruházat      3.1   
4        105   Németország    Olaszország       Élelmiszer      8.7   

   szallitasi_ido_nap    suly_lb  gyors_szallitas  
0                  14  11.464024            False  
1                   4   3.306930             True  
2                  21  28.219136            False  
3                  18   6.834322            False  
4                   6  19.180194             True

---